app/faceExpressions/conf/emotions_rfe.ini

```diff
@@ -2,19 +2,19 @@
 emotions 10
 
 //configuration of facial expressions
-//index name Left-Eyebrow(HEX) Right-Eyebrow(HEX) Mouth(HEX) Eyelids(% open)
+//index name Left-Eyebrow(HEX) Right-Eyebrow(HEX) Mouth(HEX) Eyelids(% open, e.g 00=0%, 05=50%, 10 = 100%)
 //index starts at 1
 
-E1   neu 00h 00h 00h 38h
-E2   ta1 **h **h 00h **h
-E3   ta2 **h **h 03h **h
-E4   hap 01h 01h 01h 70h
-E5   sad 02h 02h 02h 38h
-E6   sur 03h 03h 03h 38h
-E7   ang 04h 04h 04h 70h
-E8   evi 05h 05h 05h 70h
-E9   shy 06h 06h 06h 70h
-E10  cun 07h 07h 07h 38h
+E1   neu 00h 00h 00h 00p
+E2   ta1 **h **h 00h **p
+E3   ta2 **h **h 03h **p
+E4   hap 01h 01h 01h 10p
+E5   sad 02h 02h 02h 00p
+E6   sur 03h 03h 03h 00p
+E7   ang 04h 04h 04h 10p
+E8   evi 05h 05h 05h 10p
+E9   shy 06h 06h 06h 10p
+E10  cun 07h 07h 07h 00p
```

Comment:
Wouldn't it be better to use 3 digits to represent percentages so that we could have 100 --> 100%?
Is there any limitation in the number of digits?

Reply:
The third character it is ignored in the parsing phase of the ini file, for using it I have handle in a different way the configuration phase for the old robot and the new one.
In this way the parsing part is not touched, just the execution phase
I agree that this convention is quite cryptic but in theory this ini file should be modified only by developers, and I will update the documentation explaining how this coding work as soon as I am sure about those values

Reply:
Agreed! Thanks for the exhaustive answer on this 👍

Reply:
Hi @Nicogene and @pattacini, thank you for your work!

From a user point of view, I'm also a bit confused with this encoding:

- It has always been in **hexadecimal**. I agree that a decimal encoding is more intuitive. However, I would add a check on the percentage values to bound them between 0 and 100%. For example, if a developer, that did not read the new documentation, put a 70h (the h or p is transparent) or simply use an old  `.ini` file, this will break the eyelids moving them to position 700%

- From what you write `00p = 0% opening = closed` and `10p = 100% opening`. if this is true, `neu, sad, cun, sur` emotions are going to have **closed eyelids**, which is not good.